### PR TITLE
[Issue#147]Fix the bug of threadpool shutdown may be incomplete.

### PR DESCRIPTION
**Pull request by zhang-lujie**
#147 In old code, when thread_count is zero we destroy the thread group. Then we move the destroy operation to a concentrated place to reduce delay. But we just return when thread_count is zero, which may lead to incompleted threadpool shutdown. In new code, we set shutdown flag before judging whether thread_count is zero to fix the bug.

### CODE REVIEW

**Review by nbuxrr — APPROVED** [review]

File: sql/threadpool_unix.cc [hunk 1]
```diff
@@ -1028,14 +1028,13 @@ static void thread_group_close(thread_group_t *thread_group)
   DBUG_ENTER("thread_group_close");
 
   mysql_mutex_lock(&thread_group->mutex);
+  thread_group->shutdown= true;
   if (thread_group->thread_count == 0)
```
Comment (zhang-lujie) [inline sql/threadpool_unix.cc:1032]: 等号和左值相连为了遵循当前代码块风格

**Review by zhang-lujie — COMMENTED** [review]
> Thanks for your pull request. Before we can look at your pull request, you'll need to sign a Contributor License Agreement (CLA).
> 
> The following commits have not yet signed CLA.
> 
> **[95f1d0e](https://github.com/kunpengcompute/mysql-server/commit/95f1d0e2893c4c9c1c55fd7ca2a3d1d248d5dd8f)** | Fix the bug of threadpool shutdown may be incomplete.
> 
> 📝 **Please access [here](https://clasign.osinfra.cn/sign/Z2l0aHViJTJGa3VucGVuZ2NvbXB1dGU=) to sign the CLA.**
> 
> It may take a couple minutes for the CLA signature to be fully registered; after that, please reply here with a new comment: **/check-cla** to verify. Thanks.
> 
> * If you've already signed a CLA, it's possible you're using a different email address for your gitee account. Check your existing CLA data and verify the email.
> * If you signed the CLA as an employee or a member of an organization, please contact your corporation or organization to verify you have been activated to start contributing.
> * If you have done the above and are still having issues with the CLA being reported as unsigned, please feel free to file an issue.

/check-cla

**Comment by zhang-lujie** [comment]
> Thanks for your pull request. Before we can look at your pull request, you'll need to sign a Contributor License Agreement (CLA).
> 
> The following commits have not yet signed CLA.
> 
> **[95f1d0e](https://github.com/kunpengcompute/mysql-server/commit/95f1d0e2893c4c9c1c55fd7ca2a3d1d248d5dd8f)** | Fix the bug of threadpool shutdown may be incomplete.
> 
> 📝 **Please access [here](https://clasign.osinfra.cn/sign/Z2l0aHViJTJGa3VucGVuZ2NvbXB1dGU=) to sign the CLA.**
> 
> It may take a couple minutes for the CLA signature to be fully registered; after that, please reply here with a new comment: **/check-cla** to verify. Thanks.
> 
> * If you've already signed a CLA, it's possible you're using a different email address for your gitee account. Check your existing CLA data and verify the email.
> * If you signed the CLA as an employee or a member of an organization, please contact your corporation or organization to verify you have been activated to start contributing.
> * If you have done the above and are still having issues with the CLA being reported as unsigned, please feel free to file an issue.

/check-cla

**Comment by zhang-lujie** [comment]
> Thanks for your pull request. Before we can look at your pull request, you'll need to sign a Contributor License Agreement (CLA).
> 
> The following commits have not yet signed CLA.
> 
> **[95f1d0e](https://github.com/kunpengcompute/mysql-server/commit/95f1d0e2893c4c9c1c55fd7ca2a3d1d248d5dd8f)** | Fix the bug of threadpool shutdown may be incomplete.
> 
> 📝 **Please access [here](https://clasign.osinfra.cn/sign/Z2l0aHViJTJGa3VucGVuZ2NvbXB1dGU=) to sign the CLA.**
> 
> It may take a couple minutes for the CLA signature to be fully registered; after that, please reply here with a new comment: **/check-cla** to verify. Thanks.
> 
> * If you've already signed a CLA, it's possible you're using a different email address for your gitee account. Check your existing CLA data and verify the email.
> * If you signed the CLA as an employee or a member of an organization, please contact your corporation or organization to verify you have been activated to start contributing.
> * If you have done the above and are still having issues with the CLA being reported as unsigned, please feel free to file an issue.

/check-cla

**Comment by zhang-lujie** [comment]
> Thanks for your pull request. Before we can look at your pull request, you'll need to sign a Contributor License Agreement (CLA).
> 
> The following commits have not yet signed CLA.
> 
> **[95f1d0e](https://github.com/kunpengcompute/mysql-server/commit/95f1d0e2893c4c9c1c55fd7ca2a3d1d248d5dd8f)** | Fix the bug of threadpool shutdown may be incomplete.
> 
> 📝 **Please access [here](https://clasign.osinfra.cn/sign/Z2l0aHViJTJGa3VucGVuZ2NvbXB1dGU=) to sign the CLA.**
> 
> It may take a couple minutes for the CLA signature to be fully registered; after that, please reply here with a new comment: **/check-cla** to verify. Thanks.
> 
> * If you've already signed a CLA, it's possible you're using a different email address for your gitee account. Check your existing CLA data and verify the email.
> * If you signed the CLA as an employee or a member of an organization, please contact your corporation or organization to verify you have been activated to start contributing.
> * If you have done the above and are still having issues with the CLA being reported as unsigned, please feel free to file an issue.

/check-cla

**Comment by zhang-lujie** [comment]
> Thanks for your pull request. Before we can look at your pull request, you'll need to sign a Contributor License Agreement (CLA).
> 
> The following commits have not yet signed CLA.
> 
> **[95f1d0e](https://github.com/kunpengcompute/mysql-server/commit/95f1d0e2893c4c9c1c55fd7ca2a3d1d248d5dd8f)** | Fix the bug of threadpool shutdown may be incomplete.
> 
> 📝 **Please access [here](https://clasign.osinfra.cn/sign/Z2l0aHViJTJGa3VucGVuZ2NvbXB1dGU=) to sign the CLA.**
> 
> It may take a couple minutes for the CLA signature to be fully registered; after that, please reply here with a new comment: **/check-cla** to verify. Thanks.
> 
> * If you've already signed a CLA, it's possible you're using a different email address for your gitee account. Check your existing CLA data and verify the email.
> * If you signed the CLA as an employee or a member of an organization, please contact your corporation or organization to verify you have been activated to start contributing.
> * If you have done the above and are still having issues with the CLA being reported as unsigned, please feel free to file an issue.

/check-cla

**Pull request closed without merging**